azurelinuxagent/common/protocol/wire.py

```diff
@@ -781,76 +782,67 @@ def update_goal_state(self, force_update=False):
         Updates the goal state if the incarnation or etag changed or if 'force_update' is True
         """
         try:
-            updated = False
-
             goal_state = GoalState(self)
 
             # Always update the hostgaplugin, since the agent may issue requests to it even if there are no other changes
             # in the goal state (e.g. report status)
             self._update_host_plugin(goal_state.container_id, goal_state.role_config_name)
 
+            # Fetch the full goal state if needed
             if force_update:
                 logger.info("Forcing an update of the goal state..")
 
             fetch_full_goal_state = force_update or \
                 self._goal_state is None or self._extensions_goal_state is None or \
                 self._goal_state.incarnation != goal_state.incarnation
 
-            #
-            # TODO: Today we always update self._extensions_goal_state using the ExtensionsConfig. After that, (unless
-            #       FastTrack is disabled in waagent.conf) we query the vmSettings only to compare them against the
-            #       ExtensionsConfig.
-            #       When FastTrack is fully implemented, we will need instead to update self._extensions_goal_state
-            #       from the vmSettings and fallback to the ExtensionsConfig only if FastTrack is disabled or if it is
-            #       not supported by the HostGAPlugin.
-            #       Extensions processing is always based on self._extensions_goal_state, while self._extensions_goal_state_from_vm_settings
-            #       is currently held only for debugging purposes.
-            #
-            if fetch_full_goal_state:
+            if not fetch_full_goal_state:
+                goal_state_updated = False
+            else:
                 goal_state.fetch_full_goal_state(self)
                 self._goal_state = goal_state
-                if self._goal_state.extensions_config_uri is None:
-                    self._extensions_goal_state = ExtensionsGoalStateFactory.create_empty()
-                else:
-                    xml_text = self.fetch_config(self._goal_state.extensions_config_uri, self.get_header())
-                    self._extensions_goal_state = ExtensionsGoalStateFactory.create_from_extensions_config(goal_state.incarnation, xml_text, self)
-                updated = True
-
-            if conf.get_enable_fast_track():
-                # Since currently we query vmSettings only to exercise the HostGAPlugin, errors in this section of the code should not
-                # prevent the agent from processing the goal state; we simply report a summary of those errors.
+                goal_state_updated = True
```

Comment:
maybe rename this to `fabric_goal_state_updated` for more clarity?

Reply:
the 2 goal states are named _goal_state and _vm_settings_goal_state, the update variables are parallel to those names (goal_state_updated and vm_settings_goal_state updated)

renaming self._goal_state to self.fabric_goal_state would not be appropriate, since in the next iteration there will be only 1 goal state (which can come from fabric or fast track) and self._vm_settings_goal_state will go away)

Reply:
sounds good.